### PR TITLE
Refactor TensorFlow operator into base and Triton runtime operator

### PR DESCRIPTION
**Pull request by oliverholworthy**
Extract Triton parts of `PredictTensorflow` into a Triton-specific operator and integrate with the Triton runtime.

### CODE REVIEW

File: merlin/systems/dag/ops/operator.py [hunk 1]
```diff
@@ -288,12 +290,13 @@ def export(
 
         os.makedirs(node_export_path, exist_ok=True)
         os.makedirs(os.path.join(node_export_path, str(version)), exist_ok=True)
-        copyfile(
-            os.path.join(
-                os.path.dirname(__file__), "..", "..", "triton", "models", "oprunner_model.py"
-            ),
-            os.path.join(node_export_path, str(version), "model.py"),
-        )
+        with importlib.resources.path(
```
Comment (karlhigley) [inline merlin/systems/dag/ops/operator.py:293]: TIL 👍🏻

**Review by karlhigley — APPROVED** [review]

File: merlin/systems/triton/models/executor_model.py [hunk 2]
```diff
@@ -119,7 +119,8 @@ def execute(self, requests):
                 tb_string = repr(traceback.extract_tb(exc_traceback))
                 responses.append(
                     pb_utils.InferenceResponse(
-                        tensors=[], error=f"{exc_type}, {exc_value}, {tb_string}"
+                        output_tensors=[],
```
Comment (oliverholworthy) [inline merlin/systems/triton/models/executor_model.py:122]: This change corrects the arguments passed to `InferenceResponse` which makes the error messages a bit clearer when something goes wrong. Previously errors were all wrapped in a message about arguments to the `InferenceResponse` which made things more difficult to quickly scan and see what was going wrong:

```
Failed to process the request(s) for model instance '1_predictforest', message: TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
E           1. c_python_backend_utils.InferenceResponse(output_tensors: List[c_python_backend_utils.Tensor], error: c_python_backend_utils.TritonError = None)
E
E       Invoked with: kwargs: tensors=[], error="'...
```

### Example

**Before**

image


**After**

image

**Comment by oliverholworthy** [comment]
rerun tests

**Comment by oliverholworthy** [comment]
rerun tests

**Comment by karlhigley** [comment]
rerun tests

**Comment by jperez999** [comment]
rerun tests

**Comment by jperez999** [comment]
rerun tests

**Comment by jperez999** [comment]
rerun tests

**Comment by jperez999** [comment]
rerun tests

**Comment by jperez999** [comment]
rerun tests

**Comment by jperez999** [comment]
rerun tests

**Comment by karlhigley** [comment]
@jperez999 tells me that the main thread of PyTest grabs half the GPU memory via `configure_tensorflow` when running the examples, and then grabs another half when running the normal tests.

**Comment by jperez999** [comment]
rerun tests

**Comment by jperez999** [comment]
rerun tests

**Pull request merged**